C/services/south/ingest.cpp

```diff
@@ -1056,6 +1056,15 @@ void Ingest::useFilteredData(OUTPUT_HANDLE *outHandle,
 	lock_guard<mutex> guard(ingest->m_useDataMutex);
 	
 	vector<Reading *> *newData = readingSet->getAllReadingsPtr();
+	if (!ingest->m_data)
+	{
+		// If we are called during shutdown there will be no m_data in place
+		// and we create a new one to handle this special case. In this case
+		// the m_data will not be explicitly deleted. However as we are shutting
+		// down this will note cause a problem as all memory is recovered at process
```

Comment:
"note" -> "not"

Is it possible that pipeline shutdown happens without the service going down? E.g. removal of all filters. It is still an corner case, and we may leave it unhandled if it is not straightforward to handle this.

Reply:
Yes, it is possible. I will need to think about that one. My inclination is to checkin what we have, as it is not going to result in a recurring memory leak, and then resolve this afterwards. That way we can at least progress with this and the FOGL-8944 that is probably a bigger issue than this.

Reply:
Thanks. I agree.